pages/products/[slug].tsx

```diff
@@ -235,10 +268,55 @@ const ProductPage = ({
               >
                 {loadingAddToCheckout ? "Adding..." : "Add to cart"}
               </button>
-            ) : (
+            ) : null}
+
+            {!selectedVariant && (
+              <p className="text-lg- text-yellow-600">
+                Please choose a variant
+              </p>
+            )}
+
+            {selectedVariant?.quantityAvailable === 0 && (
               <p className="text-lg- text-yellow-600">Sold out!</p>
             )}
+
             {!!addToCartError && <p>{addToCartError}</p>}
+
+            {product?.description && (
+              <div className="text-base text-gray-700 space-y-6">
+                <RichText jsonStringData={product.description} />
+              </div>
+            )}
+
+            {attributes.length > 0 && (
+              <div>
+                <p className="text-lg mt-2 font-medium text-gray-500">
+                  Attributes
+                </p>
+                <div className="">
+                  {attributes.map((attribute: any) => (
+                    <div
+                      key={attribute.attribute.name}
+                      className="grid grid-cols-2"
+                    >
+                      <div>
+                        <p>{attribute.attribute.name}</p>
+                      </div>
+                      <div>
+                        {attribute.values.map((value: any, index: number) => (
```

Comment:
```suggestion
                        {attribute.values.map((value, index) => (
```
TS should infer types of attributes.